### PR TITLE
feat: popup panel stacking

### PR DESCRIPTION
**Pull request by LPLafontaineB**
### Description (*)

This PR simplifies PopupPanel so its only responsibility is to display a single message. It moves the responsibility of creating those messages to the PopupManager. The PopupManager will instantiate PopupPanel gameObjects and display them when it is called. It will reuse those instantiated objects when possible (which is most of the time since we likely won't display more than one or two at once in a regular case). 
### Issue Number(s) (*)

Fixes issue(s): [MTT2733](https://jira.unity3d.com/browse/MTT-2733)
### Manual testing scenarios

The simplest way to see this PR's effects is to duplicate calls to ShowPopupPanel() to have multiple popups at once. Or add a coroutine that perdically calls this, so that you can experiment with closing some of them before new ones are created.
### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by LPLafontaineB** [comment]
Put on hold as it is not needed for GDC

**Comment by LPLafontaineB** [comment]
tested on my phone with a 20:9 ratio and 720x1600 resolution (both in portrait and landscape orientations) with enough popups stacked to reach the maximum popup offset and they all were still visible in the screen.

File: Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs:19]: not a fan of implicit meaning like this. (canvasGroup alpha == "IsDisplaying --> what happens when we change our structure and canvasGroup isn't representative of the whole popup anymore?
Having an explicit IsDisplaying = false in Hide() would make this clearer

**Review by SamuelBellomo — APPROVED** [review]

File: Assets/BossRoom/Scripts/Client/UI/PopupManager.cs
Comment (pdeschain) [inline Assets/BossRoom/Scripts/Client/UI/PopupManager.cs:19]: This is a good candidate for being turned into singleton with our DI.

**Comment by pdeschain** [comment]
Apart from the DI comment - this PR is 👍

File: Assets/BossRoom/Scripts/Client/UI/PopupManager.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/UI/PopupManager.cs:19]: Since this is defined in the Client assembly, I cannot bind it inside ApplicationController, however, so it should either be moved to the Shared assembly or it should be bound in all the ClientGameStates. If the latter makes more sense, it would need to wait until PR #579 is merged to avoid conflicts

**Review by fernando-cortez — COMMENTED** [review]
Will approve once the conversation regarding DI is resolved.

**Review by SamuelBellomo — APPROVED** [review]

**Review by SamuelBellomo — APPROVED** [review]

**Review by pdeschain — APPROVED** [review]

**Pull request merged**
